Add size test for minimal emscripten::val demo

RReverser opened the pull request:
Noticed we don't have any size tests for it so adding a minimal one to track improvements/regressions to emscripten::val.

RReverser commented:
I don't get why CI gets different sizes than me all of the sudden.

Last time I ran into something like this was due to Closure update that I didn't have, but now that we have `./bootstrap` check (and I ran it manually just in case again) that shouldn't happen anymore. And yet, CI gets 3 bytes difference.

RReverser commented:
Huh, running my own test (which was authored on Windows) inside WSL results in yet another size, different both from my Windows one and the CI one:

```
size of a.html == 673, expected 673, delta=0
size of a.html.gz == 431, expected 431, delta=0
size of a.js == 7400, expected 7400, delta=0
size of a.js.gz == 3091, expected 3091, delta=0
size of a.wasm == 8967, expected 8958, delta=9 (+0.10%)
size of a.wasm.gz == 4604, expected 4612, delta=-8 (-0.17%)
Total output size=17040 bytes, expected total size=17031, delta=9 (+0.05%)
Total output size gzipped=8126 bytes, expected total size gzipped=8134, delta=-8 (-0.10%)
Oops, overall generated code size regressed by 9 bytes!
```

There has to be difference in something like newlines somewhere...

RReverser commented:
Ahhh I remembered one of the other reasons now. It's because my emsdk is behind tot. @sbc100 Probably something `./bootstrap` should be checking as well?

sbc100 commented on test/test_other.py:
```diff
@@ -10386,7 +10386,9 @@ def test_minimal_runtime_code_size(self, test_name, js, compare_js_output=False)
-                          '-sDYNAMIC_EXECUTION=0'
+                          '-sDYNAMIC_EXECUTION=0',
+                          '-lc++',
+                          '-lc++abi'
```
Why are these two libraries needed? They should be automatically included by the compiler.

RReverser replied:
I'm not sure but I think it's because we use `-sSTRICT`. Without it I'd get linking errors for C++ builtins, and I think I've seen the same in the past when using `STRICT`.

sbc100 replied:
Ah.. in that case the fix is to use EMXX below instead of EMCC.

RReverser replied:
Ah so it's not the AUTO_NATIVE_LIBRARIES that's causing the issue (I thought libc++ was counted among them) but the default C vs C++.

Is it safe to change it to EMXX even for other tests or does it have to be only for Embind?

sbc100 replied:
Ideally C tests should use `emcc` and C++ tests should use `em++`.

sbc100 replied:
For most tests this is already automatic.  See https://github.com/emscripten-core/emscripten/blob/054bf307ee8391ffea9037761e3c9e2bc54dcbbe/test/common.py#L946

In this case its enough to simply switch to EMXX I think.

sbc100 replied:
Alternatively you could use `self.emcc` or `self.build` instead of directly using `self.run_process`

RReverser replied:
It's a bit confusing because e.g. in `hello_webgl_sources` there are two filenames, one is C++ and another is C, and modifying test means I modify it for all those testcases.

I guess I can just split out first argument and assume that's what should be used as a filename for now...

RReverser replied:
OTOH switching to EMXX seems easier.

RReverser replied:
Ok no I tried those different options and I'm just seeing different errors each time, mostly due to that mix of C and C++ in one invocation in hello_webgl_sources. I believe the way that test is done is not truly correct if we want to use different compilers for different langs, since technically each source would have to be compiled with `emcc`/`em++` respectively and then linked together... but I don't want to take on fixing that as part of this PR.

I'd rather keep the current workaround until someone refactors the parent test to account for different languages.

sbc100 commented:
> I don't get why CI gets different sizes than me.
> 
> Last time I ran into something like this was due to Closure update that I didn't have locally, but now that we have `./bootstrap` check (and I ran it manually just in case again) that shouldn't happen anymore. And yet, CI gets 3 more bytes than me locally.

Aside from installing `tot` the only other reason I can think of is that you maybe didn't do `--clear-cache`, but I think you know about that one.

RReverser commented:
> maybe didn't do `--clear-cache`, but I think you know about that one.

You'd be surprised... I did `./embuilder build sysroot --force` as aside from headers I'm used to relying on Emscripten rebuilding cache automatically when switching branches, but apparently that wasn't enough in this case as something else needed rebuild too. Clearing cache added those 3 bytes back.

RReverser commented:
Ok I finally found an elegant solution for `hello_webgl` not fitting into either compiler because it uses a mix of C and C++ files: I just changed the C++ file to C, since it already was entirely C-like with 2 minor incompatibilities.

I fixed those so now all size tests can run with respective `compiler_for` that is based on the first source file.

Please take another look.
